pyvista/core/filters/image_data.py

```diff
@@ -772,3 +773,114 @@ def _flip_uniform(self, axis) -> 'pyvista.ImageData':
+        smoothing: bool = False,
+        smoothing_num_iterations: int = 50,
+        smoothing_relaxation_factor: float = 0.5,
+        smoothing_constraint_distance: float = 1,
+        output_mesh_type: str = 'quads',
+        output_style: str = 'default',
+        scalars: Optional[str] = None,
+        progress_bar: bool = False,
+    ) -> 'pyvista.PolyData':
+        """Generate labeled contours from 3D label maps.
+
+        SurfaceNets algorithm is used to extract contours preserving sharp
+        boundaries for the selected labels from the label maps.
+        Optionally, the boundaries can be smoothened to reduce the staircase
+        appearance in case of low resolution input label maps.
+
+        This filter requires that the :class:`ImageData` has integer point
+        scalars, such as multi-label maps generated from image segmentation.
+
+
+        Parameters
+        ----------
+        n_labels : int, optional
+            Number of labels to be extracted (all are extracted if None is given).
+
+        smoothing : bool, default: False
+            Apply smoothing to the meshes.
+
+        smoothing_num_iterations : int, default: 50
+            Number of smoothing iterations.
+
+        smoothing_relaxation_factor : float, default: 0.5
+            Relaxation factor of the smoothing.
+
+        smoothing_constraint_distance : float, default: 1
+            Constraint distance of the smoothing.
+
+        output_mesh_type : str, default: 'quads'
+            Mesh type - triangles or quads.
+
+        output_style : str, default: 'default'
+            Output style (default, boundary). "selected" is currently not implemented.
+
+        scalars : str, optional
+            Name of scalars to process. Defaults to currently active scalars.
+
+        progress_bar : bool, default: False
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.PolyData
+            :class:`pyvista.PolyData` Labeled mesh with the segments labeled.
+
+        References
+        ----------
+        Sarah F. Frisken, SurfaceNets for Multi-Label Segmentations with Preservation
+        of Sharp Boundaries, Journal of Computer Graphics Techniques (JCGT), vol. 11,
+        no. 1, 34-54, 2022. Available online http://jcgt.org/published/0011/01/03/
+
+        https://www.kitware.com/really-fast-isocontouring/
+
+        Examples
+        --------
+        See :ref:`contouring_example` for a full example using this filter.
+        """
+        if not hasattr(_vtk, 'vtkSurfaceNets3D'):  # pragma: no cover
+            from pyvista.core.errors import VTKVersionError
+
+            raise VTKVersionError('Surface nets 3D require VTK 9.3.20230807rc0 or newer.')
+
+        alg = _vtk.vtkSurfaceNets3D()
+        if scalars is None:
+            set_default_active_scalars(self)  # type: ignore
+            field, scalars = self.active_scalars_info  # type: ignore
+            if field.value == 1:
```

Comment:
Use the enums from 

https://github.com/pyvista/pyvista/blob/94c484c6a6a54b86704168a54666323234d7fdf5/pyvista/core/utilities/arrays.py#L15-L21

Reply:
We should also use `field != FieldAssociation.POINT`, or similar.  It is probably certain that the only other possible value is cells here, but better to be safe and it is more understandable in the code.

Reply:
Thanks for highligting this enum for me, fully agree!

I copied the logic from gaussian_smooth and image_dilate_erode in the same file. Shall I update these two functions as well with the comparison against the enum for consistency (despite it being unrelated to this PR)?

Reply:
This is indeed ballooning quite a bit (9 test failures). I will not add it to this PR.